### PR TITLE
Update nn_ops.py

### LINKED ISSUE #46834: floating point exception in tf.nn.avg_pool3d and tf.nn.max_pool3dwhen ksize=0
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
floating point exception in `tf.nn.avg_pool3d` and `tf.nn.max_pool3d` when `ksize`=0

**Describe the expected behavior**
expect no crash
**Standalone code to reproduce the issue**

The code crashes in nightly version too. Check out the [gist](https://colab.research.google.com/drive/10O4Qn2S4uW-40jLDHvlUBN_vlMsQkUo-?usp=sharing)
~~~python
tf.nn.avg_pool3d(input=tf.ones((1,1,1,1,1)), strides=1, ksize=0, padding='VALID')
~~~

~~~python
tf.nn.max_pool3d(input=tf.ones((1,1,1,1,1)), strides=1, ksize=0, padding='VALID')
~~~
Output:
~~~python
Floating point exception (core dumped)
~~~

### PR DESCRIPTION
**Pull request by around-star**
Fix [#46834](https://github.com/tensorflow/tensorflow/issues/46834) and [#46994](https://github.com/tensorflow/tensorflow/issues/46994)
On having kernel size as 0 , the code crashes on execution. This pull request will raise a **FloatingPointError** upon getting a value of 0 for the `ksize` argument in `nn_ops`.

### CODE REVIEW

**Comment by allenlavoie** [comment]
Probably better to check this in the kernel itself, since that catches more uses.

**Comment by around-star** [comment]
@allenlavoie The added code itself checks the kernel and would return a `FloatingPointError` on getting its value as 0, if this is what you meant. This would add the functionality to all the nn functions/ops which gets the `ksize` argument as 0 and is not restricted to a particular operation/function.

**Comment by allenlavoie** [comment]
I mean checking in the C++ tensorflow::OpKernel itself rather than the
Python binding, returning a bad status instead of throwing an FPE. That
will cover tf.raw_ops.* usage from Python, and C++ usage, Java usage, etc.
and make sure people's SavedModels throw a sensible error.

**Comment by captain-pool** [comment]
> I mean checking in the C++ tensorflow::OpKernel itself rather than the Python binding, returning a bad status instead of throwing an FPE. That will cover tf.raw_ops.* usage from Python, and C++ usage, Java usage, etc. and make sure people's SavedModels throw a sensible error.
> […](#)

This makes sense. @around-star , here's the OpFunction where you should implement the fix. :)

https://github.com/tensorflow/tensorflow/blob/dec8e0b11f4f87693b67e125e67dfbc68d26c205/tensorflow/compiler/tf2xla/kernels/pooling_ops.cc#L77-L99

**Comment by around-star** [comment]
Hey Allen. I am Hiran Sarkar, 2021 GSOC aspirant. Currently working on the
issue #46798 <https://github.com/tensorflow/tensorflow/issues/46834> (pull
request sent #46838 <https://github.com/tensorflow/tensorflow/pull/46838>). I
built tensorflow from source on docker and added the ksize check for the
XLA kernel (in tf2xla
<https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/pooling_ops.cc>),
but it fails to work. Can you help guide me in solving this issue for all
the kernels including XLA, GPU, CPU, MKL?

**Review by allenlavoie — APPROVED** [review]
